src/test/java/org/minions/devfund/marylin/SquarelotronTest.java

```diff
@@ -0,0 +1,566 @@
+package org.minions.devfund.marylin;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * SquarelotronTest class.
+ */
+public class SquarelotronTest {
+    private static final Logger LOGGER = Logger.getLogger(WhackAMole.class.getName());
+    private static final int ZERO = 0;
+    private static final int ONE = 1;
+    private static final int TWO = 2;
+    private static final int THREE = 3;
+    private static final int FOUR = 4;
+    private static final int FIVE = 5;
+    private static final String EMPTYSPACE = " ";
+    private static final String INITIALWORD = "INITIAL";
+    private static final String ACTUALWORD = "ACTUAL";
+    private static final String EXPECTEDWORD = "EXPECTED";
+    private static final String ASSERTIONMESSAGE = "Grid is not the expected.";
+
+    //Initialize squarelotrons with different sizes
+    private Squarelotron initialSquarelotronTwoByTwo = new Squarelotron(TWO);
+    private Squarelotron initialSquarelotronThreeByThree = new Squarelotron(THREE);
+    private Squarelotron initialSquarelotronFourByFour = new Squarelotron(FOUR);
+    private Squarelotron initialSquarelotronFiveByFive = new Squarelotron(FIVE);
+
+    /**
+     * Method to print a grid.
+     *
+     * @param grid  grid to print.
+     * @param title title to show.
+     */
+    void printGrid(final int[][] grid, final String title) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(title).append(" - ");
+        for (int i = 0; i < grid.length; i++) {
+            for (int j = 0; j < grid.length; j++) {
+                stringBuilder.append(EMPTYSPACE).append(grid[i][j]).append(EMPTYSPACE);
+            }
+        }
+        LOGGER.log(Level.INFO, stringBuilder.toString());
+    }
+
+    /**********************************************************
+     **********************upsideDownFlip**********************
+     *********************************************************/
+
+    /**
+     * {{1, 2}, {3, 4}}.
+     */
+    @Test
+    public void testUpsideDownFlipTwoByTwoGridFirstRing() {
+        final int[][] expectedGrid = {{3, 4}, {1, 2}};
+        printGrid(initialSquarelotronTwoByTwo.getSquarelotron(), INITIALWORD);
+        printGrid(expectedGrid, EXPECTEDWORD);
+        Squarelotron actualSquarelotron = initialSquarelotronTwoByTwo.upsideDownFlip(ONE);
+        printGrid(actualSquarelotron.getSquarelotron(), ACTUALWORD);
+        assertTrue(ASSERTIONMESSAGE, Arrays.deepEquals(expectedGrid, actualSquarelotron.getSquarelotron()));
```

Comment:
Its better to use assertArrayEquals.
```java
assertArrayEquals(ASSERTIONMESSAGE, expectedGrid, actualSquarelotron.getSquarelotronMatrix());
```
Internally the method calls Arrays.deepEquals.
You should replace in all your tests.

Reply:
done